### DISCUSSION TITLE
[Feature Request] LDAP/Oauth Support

### DISCUSSION THREAD

**Discussion by maxfromwtf** [linuxserver/Heimdall#681] [Ideas]
I could see this being a project that could be implemented with something like Keycloak and used as a website user home page to be redirected from after signing in.

**Comment by NemesisRE** [comment]
That would be pretty nice I use [authentik](https://goauthentik.io/) as my sso auth provider at home and it would be nice to authenticate against that.

I am not a dev but as far as I can tell, all we need should be in https://github.com/laravel/socialite and https://socialiteproviders.com/

As providers I could think of
- [keycloak](https://socialiteproviders.com/Keycloak/)
- [saml2](https://socialiteproviders.com/Saml2)
these two are common and can be selfhosted maybe there are others that I forgot

**Comment by lknite** [comment]
let's do it

**Comment by lknite** [comment]
Hey, I think this may already exist by configuring laravel.  I'm not quite sure how to get it working though.  Maybe one of you guys might know?  I'll keep working on it.

Created an issue asking about this:  https://github.com/linuxserver/Heimdall/issues/899

**Comment by lknite** [comment]
Also created another discussion with the SSO idea a little more fleshed out:  https://github.com/linuxserver/Heimdall/discussions/898

**Comment by cpoulter-kentcorp** [comment]
I have been hoping for saml2 implementation for months for heimdall. I would like to roll this out to my team and be able to deploy a stock set of definitions, but still allow them to add their own stuff, visible to either just themselves or all users

**Comment by MHagreis** [comment]
please add this asap
